ls1intum/Artemis
# [Bugfix] Fix grading related sentry issues

### Checklist

### Motivation and Context



This fixes several sentry issues and bugs.

Fixes #2409 
Fixes #2410 

Also fixes [this sentry issue](https://sentry.io/organizations/ls1intum/issues/1741295727/?environment=prod&project=1440029&query=is%3Aunresolved&sort=freq&statsPeriod=7d)

### Description


- Added null / undefined checks
- In the grading page, the input values are not checked for NaN / null values and given a default value.

### Steps for Testing


1. Log in to Artemis
2. Navigate to the Grading Page of a Programming Exercise
3. Try to add NaN values (letters, symbols, etc) -> When navigating out of the input field, the value should be reset to the previous value.
4. Try empty values -> When navigating out of the input field, the value should be reset to the default value for this field.

### Test Coverage




* programming-exercise-configure-grading.component.ts: 82%

### Screenshots

![ezgif com-gif-maker (1)](https://user-images.githubusercontent.com/13920539/99669228-0e9f7600-2a6f-11eb-9452-56f76958564c.gif)
